rfc/0011-value-transformer/README.md

```diff
@@ -0,0 +1,191 @@
+<!--
+SPDX-FileCopyrightText: 2023 Friedrich-Alexander-Universitat Erlangen-Nurnberg
+
+SPDX-License-Identifier: AGPL-3.0-only
+-->
+
+# RFC 0000: Value Transformer
+
+| | |
+|---|---|
+| Feature Tag | `value-transformer` |
+| Status | `DRAFT` | <!-- Possible values: DRAFT, DISCUSSION, ACCEPTED, REJECTED -->
+| Responsible | `georg-schwarz` |
+<!-- 
+  Status Overview:
+  - DRAFT: The RFC is not ready for a review and currently under change. Feel free to already ask for feedback on the structure and contents at this stage.
+  - DISCUSSION: The RFC is open for discussion. Usually, we open a PR to trigger discussions.
+  - ACCEPTED: The RFC was accepted. Create issues to prepare implementation of the RFC.
+  - REJECTED: The RFC was rejected. If another revision emerges, switch to status DRAFT.
+-->
+
+## Summary
+
+This RFC introduces the concept of Value Transformers that allow reading values, writing values, and transforming values.
+
+**Note:** This RFC focuses enhancing the current state but has an elaborate "Possible Enhancements" section that showcases how this concept can be extended in the future.
+
+## Motivation
+
+Right now, we can only "parse" values but not manipulate them in order to clean data.
+
+## Explanation
+
+```
+transformer <name> {
```

Comment:
The syntax is very close to a block, I don't think that is a good idea.

Can we consider `from`/`to` for input and output?

Also, is a transformer just the singular M2 concept? We might want to implement custom transformers on a language level, I'd consider making this one type of transformer, an `ExpressionTransformer` for example. Combined, this would become:

```
transformer <name> oftype ExpressionTransformer {
  from <inputName> oftype <valuetype>; // potentially many
  to <outputName> oftype <valuetype>; // for now: exactly one

  <outputName>: <expression producing output valuetype>;
}

Reply:
Good point. I could imagine sth like this as well.

Reply:
Regarding composite blocks, that may be added in the future, the keywords `input` and `output` could fit there as well. In case their semantics will be the same (or similar), I'd go with `input` and `output` for both concepts. Otherwise I think `from` and `to` are fine too.

---

Not sure about the transformer type part. Would the `mapper` concept drafted below be considered a potential other type besides `ExpressionTransformer`?

Should a user be able to create own transformer types? If not, a distinction via a keyword (e.g. `transformer` and `mapper`) would be suited better in my opinion.

Reply:
Didn't even remember `from`/`to` is technically like pipes :D. I still like it more because it feels more intuitive to me for transforming something.

Re: Types, I would probably consider mappers other types of transformers. But I can see how users would not be able to add new transformer types so it would be okay to leave out the oftype (?). Just going with the transformer keyword has my blessing at least.